### DISCUSSION TITLE
Functionality allowing you to add WebCal (.ics) via URL in the form of a QR Code

### DISCUSSION THREAD

**Discussion by om4r932** [bitfireAT/icsx5#378] [Ideas]
Hello everyone,

I was wondering if it would be possible to add the ability to add calendars via QR Code, because in my life, i'm using a school calendar "Hyperplanning" that i have to update yearly.

It can generate a .ical calendar in the form of a URL or a QR code, except that this functionality is only available on PC via a website

It would be awesome if you could add this feature.

**Comment by sunkup** [comment]
You mean the QR code contains the actual ical ? Or just the URL? It works already if you use "webcal://" or "webcals://" as protocol in your URL for your QR code. The QR code scanner will open the URL link in the browser and the browser will redirect to ICSx5.

So, for example:
https://www.feiertage-deutschland.de/kalender-download/ics/schulferien-hamburg.ics
becomes 
webcals://www.feiertage-deutschland.de/kalender-download/ics/schulferien-hamburg.ics

<details>

<summary>Try it out</summary>

![image](https://github.com/user-attachments/assets/252a0940-cc0c-4ebc-ac0f-9c74d5688892)

</details>
